### DISCUSSION TITLE
problem loading data

### DISCUSSION THREAD

**Discussion by UserA12345** [TorchIO-project/torchio#824] [Q&A]
Hi,

Sorry as I am not sure I posted my question at the right place. 
I am trying to load both training and validation set with len 3 from tio.suject with 4 images of 22.5mb.
I am trying to run that on 6Gb gpu but I can"t manage to do that and I don't know where I should look to improve my code.
Please find attached my code and the error.
I am thinking about to create a specific dataset but I am not really sure about that.
Moreoever I don't understand why the error states it tried to allocate 9gb whereas my files are much smaller and the dataset is very small.
Many thanks in advance

![5](https://user-images.githubusercontent.com/58914979/154453821-41d8ec9c-61df-433f-a9d0-9c1e4edc4c27.png)
![6](https://user-images.githubusercontent.com/58914979/154453819-58567d68-a86e-47dd-a8ea-c9a5e3b9b4ad.png)
![7](https://user-images.githubusercontent.com/58914979/154453816-e3d3b5e5-b80d-418c-9a57-20484da0560f.png)

**Comment by romainVala** [comment]
the memory requiremed is related to the input shape, but much more to the exact model you used (number a model parameter, and number of output)
so try to reduce the model complexity ... (or in input size)

An other parameter is also the batch size as a batch size of 4 will take twice the memory size as a batch size of 2 ... (but ok , you already have a batch size of 1)

**Reply by UserA12345** [reply]
I have seen that it is better is total number of parameters are below 1m and this model has 808K parameters so I thought it was ok

**Comment by romainVala** [comment]
6G is a quite too low for 3DCNN ...

**Comment by fepegar** [comment]
As @romainVala said, you probably won't be able to train with full volumes on a 6-GB GPU. I suggest you use [patch-based training](https://torchio.readthedocs.io/patches/index.html).
